gowin-pll.py

```diff
@@ -0,0 +1,253 @@
+#!/usr/bin/env python3
+#
+# pll tool to find best match for the target frequency
+# calculations based on: https://github.com/juj/gowin_fpga_code_generators/blob/main/pll_calculator.html
+# limits from: http://cdn.gowinsemi.com.cn/DS117E.pdf, http://cdn.gowinsemi.com.cn/DS861E.pdf
+#
+
+import sys
+import argparse
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument(
+    "-i", "--input-freq-mhz", help="PLL Input Frequency", type=float, default=27
+)
+parser.add_argument(
+    "-o", "--output-freq-mhz", help="PLL Output Frequency", type=float, default=108
+)
+parser.add_argument("-d", "--device", help="Device", type=str, default="GW1NR-9 C6/I5")
```

Comment:
Most of our tools accept the chip indication parameter as a part number or as a series name. Having two ways is not good in itself, and we would not want to add a third way of specifying it.

Could you extract the information you need from the part number?
GW1N-UV4LQ144C6/I5
GW1N-LV1QN48C6/I5
GW1NZ-LV1QN48C6/I5
GW1NSR-LV4CQN48PC7/I6
GW1NR-LV9QN88PC6/I5